descriptor.md

```diff
@@ -0,0 +1,113 @@
+<!--[metadata]>
++++
+draft = true
++++
+<![end-metadata]-->
+
+# OpenContainers Content Descriptors
+
+OCI have several components that come to together to describe an image. 
+References between components form a [Merkle Directed Acyclic Graph (DAG)](https://en.wikipedia.org/wiki/Merkle_tree).
+The references in the _Merkle DAG_ are expressed through _Content Descriptors_.
+A _Content Descriptor_ or _Descriptor_, describes the disposition of targeted content.
+A _Descriptor_ includes the type of content, an independently-verifiable content identifier, known as a "digest" and the byte-size of the raw content.
+
+Descriptors SHOULD be embedded in other formats to securely reference external content.
+
+Descriptors MAY be used as an independent object to specify content.
+
+Implementations MUST verify fetched content against 
+
+## Properties
+
+- **`mediaType`** *string*
+
+  This REQUIRED property contains the MIME type of the referenced object.
+
+- **`digest`** *string*
+
+  This REQUIRED parameter is the digest of the content, meeting the requirements outlined in [Digests and Verification](#digests-and—verification).
+  Retrieved content SHOULD be verified against this digest before processing.
+
+- **`size`** *int*
+  This REQUIRED property specifies the size in bytes of the object.
+  This property exists so that a client will have an expected size for the content before validating.
+  If the length of the retrieved content does not match the specified length, the content should not be trusted.
+
+### Reserved
```

Comment:
On Thu, Jun 02, 2016 at 02:58:34PM -0700, Stephen Day wrote:
> > +### Reserved
> 
> Sure, but what problem does this complexity actually solve?

You get to drop this section and sleep soundly without wondering if
you should have claimed ‘urls’ (or ‘data’ [1], or …).  Vendors can add
their own optimizations without worrying about ambiguous
interpretations.  I can put an entry in my manifest saying “if you
don't want to bother with the usual layout, you can find this
directory at /ipfs/QmTkzDw…”, and clients who understand that type can
take advantage of it, while clients who don't can fall back to the
image-spec tarball layers.  Clients who are compatible with v1.0 of
the spec can *still* process v1.1 configs if that config only uses
parameters that were spec'ed in v1.0 (because the v1.1 context will
link the same definitions for those parameters).  If you want me to
keep adding to this list, let me know ;).

[1]: https://github.com/opencontainers/image-spec/pull/111/files/03d6c78ea37021b9030cf56fa97e0a0f214f18d8#r65468737